FinniversKit/Sources/Components/PromotionView/PromotionView.swift

```diff
@@ -107,33 +130,72 @@ public class PromotionView: UIView {
         titleLabel.text = viewModel.title
         primaryButton.configure(withTitle: viewModel.primaryButtonTitle)
         secondaryButton.configure(withTitle: viewModel.secondaryButtonTitle)
+        imageRatio = viewModel.image.size.width / viewModel.image.size.height
 
         if let text = viewModel.text {
             textLabel.text = text
         } else {
             textLabel.isHidden = true
         }
 
+        if let backgroundImage = viewModel.backgroundImage {
+            backgroundImageContainer.addSubview(backgroundImageView)
+            backgroundImageView.image = backgroundImage
+            backgroundImageView.fillInSuperview()
+        }
+
         imageContainer.addSubview(imageView)
         imageView.image = viewModel.image
 
         switch viewModel.imageAlignment {
         case .fullWidth:
             imageView.fillInSuperview(insets: UIEdgeInsets(top: .spacingM, leading: .spacingS, bottom: -.spacingM, trailing: -.spacingS))
         case .trailing:
-            let imageRatio = viewModel.image.size.width / viewModel.image.size.height
+            guard let imageRatio = imageRatio else { return }
             NSLayoutConstraint.activate([
                 imageView.topAnchor.constraint(equalTo: imageContainer.topAnchor),
                 imageView.trailingAnchor.constraint(equalTo: imageContainer.trailingAnchor),
                 imageView.bottomAnchor.constraint(equalTo: imageContainer.bottomAnchor),
                 imageView.widthAnchor.constraint(lessThanOrEqualTo: imageView.heightAnchor, multiplier: imageRatio),
             ])
+        case .dynamic:
+            if isCompactScreen() {
+                activateCompactConstraints()
+            } else {
+                activateRegularConstraints()
+            }
         }
 
         if let imageBackgroundColor = viewModel.imageBackgroundColor {
             imageContainer.backgroundColor = imageBackgroundColor
         }
     }
+
+    private func activateCompactConstraints() {
+        NSLayoutConstraint.deactivate(regularDynamicConstraint)
+        NSLayoutConstraint.activate(compactDynamicConstraints)
+    }
+
+    private func activateRegularConstraints() {
+        NSLayoutConstraint.deactivate(compactDynamicConstraints)
+        NSLayoutConstraint.activate(regularDynamicConstraint)
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass {
+            if isCompactScreen() {
+                activateCompactConstraints()
+            } else {
+                activateRegularConstraints()
+            }
+        }
+    }
+
+    private func isCompactScreen() -> Bool {
```

Comment:
We already have a property for this that you can use, in `UITraitCollectionExtensions.swift`. We can check `isHorizontalSizeClassRegular` instead

Reply:
Nice to know :)